### ISSUE TITLE
Captions on tables and figures

### ISSUE THREAD

**Issue by simonmika** [cogneco/typeup#3]
How should one add captions to figures and tables?

**Comment by simonmika** [comment]
What about this for a figure with comment:
`! folder/image.png This is the figure caption`
or

```
! folder/image.png
This is the figure caption
```

They should in both cases be surrounded by empty lines, like a paragraph.

**Comment by davidhesselbom** [comment]
Don't forget about alt texts for images. 

Also, floating images is something I really miss in Markdown. Any ideas on syntax for doing that?

**Comment by simonmika** [comment]
I think the alt-text could be the same as the caption.
 Do you want to have both floating images and non-floating ones in the same document?

**Comment by simonmika** [comment]
By the way, my current thinking is to use the following syntax:

```
!figure folder/image.png This is the figure caption.
```

This would also allow for things like:

```
!include folder/subcontent.tup
```

**Comment by davidhesselbom** [comment]
> Do you want to have both floating images and non-floating ones in the same document?

Yes, I'm likely going to want to have both full-width images and much more narrow images in the same document. The narrow ones I want to float, the full-width ones I'd prefer to center.

**Comment by simonmika** [comment]
Ok, I think I might have a solution.
